aws_lambda_powertools/utilities/idempotency/persistence/dynamodb.py

```diff
@@ -0,0 +1,163 @@
+import datetime
+import logging
+from typing import Any, Dict, Optional
+
+import boto3
+from botocore.config import Config
+
+from aws_lambda_powertools.utilities.idempotency import BasePersistenceLayer
+from aws_lambda_powertools.utilities.idempotency.exceptions import (
+    IdempotencyItemAlreadyExistsError,
+    IdempotencyItemNotFoundError,
+)
+from aws_lambda_powertools.utilities.idempotency.persistence.base import DataRecord
+
+logger = logging.getLogger(__name__)
+
+
+class DynamoDBPersistenceLayer(BasePersistenceLayer):
```

Comment:
__Question__: DynamoDB only supports items up to 400 KB, while Lambda supports payloads up to 6 MB. This will not be a problem for event payloads since we hash the record, but this could be on the response side. Should we document this somewhere?

Reply:
Yep, I'll add this to the docs.